### PR TITLE
ViewCollection should allow view subtypes to be defined in a way that solves circular dependency problems

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

File: go/base/static/js/src/utils.js [hunk 1]
```diff
@@ -28,9 +28,27 @@
...
+  // and an optional context to look for the object in
+  var objectByName = function(name, that) {
+    return _(name.split( '.' )).reduce(
+      function(obj, propName) { return obj[propName]; },
+      that || window);
+  };
+
+  // If given a string, gets the object by name, otherwise just returns what
+  // it was given
+  var ensureObject = function(nameOrObj, that) {
```
Comment (justinvdm) [inline go/base/static/js/src/utils.js:41]: I'm not too happy with `ensureObject`. Thoughts on a better name?

Comment (hodgestar) [inline go/base/static/js/src/utils.js:41]: What about `maybeByName`? I don't think `ensureObject` is that bad though.

**Comment by hodgestar** [comment]
:+1:. I suggested a better name for `objectName` but mostly because you asked. I'll leave it up to you to decide which of the two you prefer.

File: go/base/static/js/src/utils.js (hunk 1 above)
Comment (justinvdm) [inline go/base/static/js/src/utils.js:41]: Sounds good :)

**Pull request merged**